### DISCUSSION TITLE
programmatically switching between realtime and lazy validation

### DISCUSSION THREAD

**Discussion by goldmerc** [livewire/livewire#4381] [3. Feature ideas]
Hi, my personal favourite for validation is that the field is validated on blur, so using wire:model.lazy

However, if there is an error, I would then prefer for validation to be in realtime until the validation error is resolved.

It would be great to have an option like wire:model.lazyish which triggers realtime validation when the user blurs a field if there is an error and switches back to lazy validation once the error is resolved. 

I'm trying to figure out a solution in the meantime. The following almost works...

```
<input
    wire:model="email"
    @blur="$wire.validateProperty('email')"
    name="email"
    type="email"
/>
```

```
public function updated($propertyName)
{
    if ($this->getErrorBag()->has($propertyName)) {
        $this->validateOnly($propertyName);
    }
}

public function validateProperty($propertyName)
{
    $this->validateOnly($propertyName);
}
```
However, this is not bullet proof. If the user blurs the field by clicking directly on the submit button, it seems unpredictable which order alpine's x-on:blur and livewire's submit get called. I'm finding this can cause issues like an error message is displayed (from the submit method) and then quickly replaced with a different error message (from the x-on:blur method). 

So, I got to wondering whether I could use alpine to programmatically turn on and off lazy validation for a livewire model? Is that possible? Thanks

**Comment by joshhanley** [comment]
@goldmerc Wouldn't that defeat the purpose of using `.lazy` in the first place (reducing network requests)?

Maybe it would be better to leave the `.lazy` on the first, and instead use Alpine to hide the validation message if the user interacts with the input. Then it validates again on blur once they are finished.

What do you think?

**Comment by goldmerc** [comment]
@joshhanley thanks for the suggestion, which is a good one. It's not quite what I would like to achieve. In your example, the error disappears when the user clicks into the field. I would prefer that the error disappears when it is rectified. So, the user has a visual clue that the error has been fixed and the form will now validate.

Ideally, I would like to use .lazy but when an error is triggered, i'd like to somehow convert that field from a lazy one to realtime validation. As soon as the error is fixed, it could then return to being lazy. 

I'm basically wondering if it's possible to switch a field from lazy to realtime validation using javascript? Or even if it can somehow happen when the template is re-rendered? But I suspect the initial setup happens when the page loads and so it would probably need to happen somehow using js, if that's even possible.

Thanks for your help.

**Comment by joshhanley** [comment]
@goldmerc ah right yeah makes sense. You might be able to do something with blade conditionals and having an input with `.lazy` and another without and show the one you need when doing the live validation maybe?

**Comment by goldmerc** [comment]
@joshhanley Sorry for the slow reply. Been away for the holidays.

Just been playing with this. If I have two inputs and use blade conditionals to render one or the other, that didn't work for me. But if I have two inputs rendered and use blade conditionals to set display: hidden on the one I don't need, that almost works. The problem is as soon as the input with errors has the error resolved, the fields swap, meaning the hidden field has focus and the newly displayed field doesn't. So, the user needs to click the lazy field again to be able to enter more text. 

Is there a way to use javascript to change the wire model from lazy to real-time?

**Reply by joshhanley** [reply]
@goldmerc no worries, me too!

Hmm yeah, currently there isn't a way to change that using javascript, you'd have to hack around with adding and removing the event listeners on the input to change it.

What happened when you tried to use blade conditionals?
